feature/sessions/src/main/java/io/github/droidkaigi/confsched2023/sessions/component/FilterChipWithDropdownMenu.kt

```diff
@@ -0,0 +1,145 @@
+package io.github.droidkaigi.confsched2023.sessions.component
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FilterChip
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+
+data class SearchFilterItemUiState<T>(
+    val selectedItems: List<T>,
+    val items: List<T>,
+    val isSelected: Boolean = false,
+    val selectedValues: String = "",
+)
+
+const val FilterChipWithDropdownMenuItemTestTag = "FilterChipWithDropdownMenuItem"
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun <T> FilterChipWithDropdownMenu(
+    searchFilterItemUiState: SearchFilterItemUiState<T>,
+    onSelected: (T, Boolean) -> Unit,
+    filterChipLabel: @Composable () -> Unit,
+    dropdownMenuItemText: @Composable (T) -> Unit,
+    modifier: Modifier = Modifier,
+    filterChipLeadingIcon: @Composable (() -> Unit)? = null,
+    filterChipTrailingIcon: @Composable (() -> Unit)? = null,
+    onFilterChipClick: (() -> Unit)? = null,
+    dropdownMenuItemLeadingIcon: @Composable ((T) -> Unit)? = null,
+    dropdownMenuItemTrailingIcon: @Composable ((T) -> Unit)? = null,
+) {
```

Comment:
I applied the slots API pattern as follows:

- FilterChipLabelText
- FilterChipLeadingIcon
- FilterChipTrailingIcon
- DropdownMenuItemText
- DropdownMenuItemLeadingIcon
- DropdownMenuItemTrailingIcon

If you think the implementation intention is different, please point it out.🙇‍♂️